### ISSUE TITLE
gradient collection in HF eval step

### ISSUE THREAD

**Issue by baberabb** [EleutherAI/bergson#83]
Came across this callback for the HF `Trainer`, and wanted to confirm whether this is intended for future functionality or if I'm missing something about how it currently works, as this is [called](https://github.com/huggingface/transformers/blob/5804c1fa474f2568e359e3fb5800cb7b13fb01e2/src/transformers/trainer.py#L4578) in the `torch.no_grad()` context in `Trainer`:

https://github.com/EleutherAI/bergson/blob/8523095333a73c6acf0351d5b9f71a30f3751d27/bergson/huggingface.py#L311-L313

Either way, we should remove the hardcoded `dataset_name` assumption.
